### ISSUE TITLE
Missing function documentation in source code

### ISSUE THREAD

**Issue by shakib609** [OpenBangla/OpenBangla-Keyboard#54]
Most of the functions used inside the source code are missing documentations on how they work.
At least, some short descriptions similar to [Python Docstrings](https://www.python.org/dev/peps/pep-0257/) should be provided in the form of comments inside the source code files.
This will help contributors understand the code more easily.

**Comment by Adyel** [comment]
Yeah, without any context it is hard to understand what is what. All of the codings was done by @mominul  so he has to do it alone I guess. I can make a good looking doc website though. I think it is time to document everything properly, otherwise people can not contribute

**Comment by mominul** [comment]
আমি আগে কিছু Documentation লিখে ছিলাম। যেমন `libengine` এর `Layout.h`। কিন্তু আসলেই এখন Documentation এর যে অবস্থা, সেটা নতুন কন্ট্রিবিউটরদের জন্য দুর্বোধ্য। আমি Documentation লেখার ক্ষেত্রে তেমন ভাল নই, তাই আমার মনে হয় যদি আমরা একসাথে কাজটা করি তাহলে ভাল হয়। এক্ষেত্রে যেসব জিনিস আপনারা বুঝতে পারছেন না সেগুলো এখানে জানান। সেসব কিছু আমি এখানে বুঝিয়ে বলার চেষ্টা করব। এরপর আপনারা সেটা Documentation হিসেবে লিখে ফেলতে পারবেন।

আমি জানি আইডিয়াটি খুবই selfish :laughing: :rofl:

**Comment by Adyel** [comment]
আমরা যদি পড়ে পড়ে বুঝার চেষ্টা করি অনেক সময় লাগবে, ভুলও হবে । এটা তোমারই করতে হবে, perfect হওয়ার দরকার নেই। একটা জাস্ট কি করে এটা লিখবা । কি কি প্যারামিটার নেয়, কি রিটার্ন করে আর ফাংশনটা কি করে । Formating না হয় আমরা করলাম 
@mominul

**Comment by mominul** [comment]
আমি ভাবছি `CONTRIBUTING.md ` টাইপর কিছু একটা লেখার। এতে কন্ট্রিবিউশন করার জন্য প্রয়োজনীয় গাইড লাইন দেয়া থাকবে। ফাংশন ডকুমেন্ট করাটাও মাথায় রাখলাম :smiley: 

>  আমি কোডে অনেক Unused perameter দেখেছি, এগুলা কেন আছে জানি না ।

Unused perameter গুলো যদি চিহ্নিত করে দেয়া যায় তবে ভাল হত :wink:

**Comment by hsbasu** [comment]
> Unused perameter গুলো যদি চিহ্নিত করে দেয়া যায় তবে ভাল হত wink ।
I can do it. Actually, it is pretty simple in `vscode`.

**Comment by mominul** [comment]
Thanks for your interest, @hsbasu ! The situation has changed since this post was written, most of the code has been [rewritten in Rust](https://github.com/OpenBangla/riti). Where I tried to keep the code tidy and not leave unused parameters in the code, but you are invited to find such instances in the code if you can! :blush:
